### ISSUE TITLE
process is killled when i run txt2img.py

### ISSUE THREAD

**Issue by Yaoxingtian** [Stability-AI/stablediffusion#37]
Hi, thanks for you great job !
when i run txt2img.py, it broken and killed.  Do you know what's the problem ?

DiffusionWrapper has 865.91 M params.
making attention of type 'vanilla-xformers' with 512 in_channels
building MemoryEfficientAttnBlock with 512 in_channels...
Working with z of shape (1, 4, 32, 32) = 4096 dimensions.
making attention of type 'vanilla-xformers' with 512 in_channels
building MemoryEfficientAttnBlock with 512 in_channels...
Killed

**Comment by woctezuma** [comment]
Not enough RAM, I believe.

**Comment by sliard** [comment]
Yes, same error when I used a T4 16Go on GCP.
Same command on a A100 40Go run well.

**Comment by xiankgx** [comment]
Yeah, seems like need more than 16GB of RAM. A system with 32 GB runs fine.

**Comment by zwishenzug** [comment]
You can try this, edit the txt2img.py script.

Look for this code:

    def load_model_from_config(config, ckpt, verbose=False):
        print(f"Loading model from {ckpt}")
        pl_sd = torch.load(ckpt, map_location="cpu")
        if "global_step" in pl_sd:

Change it to:

    def load_model_from_config(config, ckpt, verbose=False):
        print(f"Loading model from {ckpt}")
        pl_sd = torch.load(ckpt, map_location="cuda")
        if "global_step" in pl_sd:

The difference here is "cuda" instead of "cpu".  This will load the model directly to the GPU (vram) instead of loading it to the CPU (system ram) first.

This should reduce the system ram requirement.

**Comment by andypsu5155** [comment]
Can you confirm if adding RAM solved the issue for you? I also have 16gbs of RAM, and a RTX 3090 and it keeps killing the process

**Comment by xiankgx** [comment]
> Can you confirm if adding RAM solved the issue for you? I also have 16gbs of RAM, and a RTX 3090 and it keeps killing the process

I can confirm this. I used AWS EC2 and changing to the next step higher instance type with more RAM (16->32) solved the issue. Or you can apply the modifications above, since not moving model weights to CPU first and directly to CUDA directly would also help with the issue.

**Comment by andypsu5155** [comment]
Thanks for the update!  I ordered more RAM for my computer, and I'm hoping once I get it installed it will work for me!
